### PR TITLE
Refactor of infrastructure, base source files, and initial logic

### PR DESCRIPTION
**Pull request by stulees**
- infrastructure revised as well as test coverage infrastructure added
- base settings and widget files added with initial logic
- README revised
- CCI revised
- removal of some previous files

### CODE REVIEW

**Comment by stulees** [comment]
@rodrigopavezi Please review, thanks. You can validate that the widget got built and the base setup is ready with this presentation: http://rva.risevision.com/#/PRESENTATION_MANAGE/id=2b68e4bb-4080-4f10-a036-b9a71d61bcb6?cid=30007b45-3df0-4c7b-9f7f-7d8ce6443013

File: circle.yml [hunk 1]
```diff
@@ -1,15 +1,31 @@
...
+    - rm google-chrome-stable.deb
     - if [ -z "$BUCKET_NAME" ]; then echo Bucket name must be specified as an environment variable; exit 1; fi
     - if [ -z $(grep version package.json |grep -o '[0-9.]*') ]; then echo Version must be specified in package.json; exit 1; fi
     - if [[ ! -d $HOME/aws ]]; then curl "https://s3.amazonaws.com/aws-cli/awscli-bundle.zip" -o "awscli-bundle.zip" && unzip awscli-bundle.zip && sudo ./awscli-bundle/install -i $HOME/aws; fi
     - npm install -g gulp
-    - npm install -g casperjs
+    - npm install -g casperjs@1.1.0-beta3
   post:
     - bower install
     - wget http://selenium-release.storage.googleapis.com/2.44/selenium-server-standalone-2.44.0.jar
```
Comment (rodrigopavezi) [inline circle.yml]: @stulees I think there is no need to run the selenium standalone. Protractor from our widget-tester should take care of that. cheers

**Comment by stulees** [comment]
@rodrigopavezi I've removed the casperjs pre-dependency in CCI, thanks

**Comment by rodrigopavezi** [comment]
@stulees looks good :+1: cheers.

**Pull request merged**